### DISCUSSION TITLE
Purpose of adding single quote to string value of literal expression

### DISCUSSION THREAD

**Discussion by omarsilva1** [eclipse-edc/Connector#2436] [General]
Hi everyone, I would like to know what the purpose is of adding a single quote to the toString method (for LiteralExpressions). I would have expected the return type to be just the value as a string or something like `"LiteralExpression: " + value` or the value wrapped around single quotes. I got confused when comparing the left expression values of policies (before I realized to cast to LiteralExpression and use `value()`), I would think others could make the same mistake.

Maybe you can tell me more, @jimmarino?
https://github.com/eclipse-edc/Connector/blob/c2bc86f0e4eafbf925e145b69ef5de9d3cd0a359/spi/common/policy-model/src/main/java/org/eclipse/edc/policy/model/LiteralExpression.java#L36-L39

**Comment by jimmarino** [comment]
Because toString() returns a representation of the expression, not its typed value. Downcasting and calling value() will return the typed value. The quote should be a wrapped single quote though:

return "'" + value + "'";
